app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -1,6 +1,25 @@
 module MfaSetupConcern
   extend ActiveSupport::Concern
 
+  def user_next_authentication_setup_path(final_path = nil)
+    case user_session[:selected_mfa_options]&.shift
```

Comment:
`shift` will mutate the session right? Should we add a `!` to indicate this might mutate?
```suggestion
  def user_next_authentication_setup_path!(final_path = nil)
    case user_session[:selected_mfa_options]&.shift
```